test/e2e/framework/node/resource.go

```diff
@@ -248,6 +249,27 @@ func GetInternalIP(node *v1.Node) (string, error) {
 	return host, nil
 }
 
+// GetAddressesByFamily returns a list of addresses of the given addressType for the given node
+// and filtered by IPFamily
+func GetAddressesByFamily(node *v1.Node, addressType v1.NodeAddressType, family v1.IPFamily) (ips []string) {
+	if family == v1.IPv6Protocol {
+		for j := range node.Status.Addresses {
```

Comment:
This can be collapsed into a single loop:
```go
for _, ndoeAddress := range node.Status.Addresses {
    if nodeAddress.Type != addressType {
        continue
    }

    nodeAddress.Address == "" {
       continue
    } 

    if family == v1.IPv6Protocol && netutil.IsIPv6String(nodeAddress.Address) {
         ips = append(ips, nodeAddress.Address)
         continue
    }

     if if family == v1.IPv4Protocol && !netutil.IsIPv6String(nodeAddress.Address) {
         ips = append(ips, nodeAddress.Address)
         continue
    }

     // error here?
}
```

Reply:
I'm not sure if I understand why we would continue on the following

```
if nodeAddress.Type != addressType {
        continue
}

if nodeAddress.Address == "" {
       continue
} 
```

The code as written has these option reversed in order to add to the returned IP slice. I rewritten in the suggested style, wouldn't we want to break on those conditions instead?

Reply:
`continue` because we skip adding IPs that either don't match the type or has an empty address. if we break then we don't check all node addresses after the first invalid one is found.

Reply:
Ok that makes sense, will change